modules/storage/src/service/mod.rs

```diff
@@ -60,10 +62,15 @@ impl TryFrom<Vec<Id>> for StorageKey {
 
 #[derive(Clone, Debug)]
 pub struct StorageResult {
-    pub key: StorageKey,
     pub digests: Digests,
 }
 
+impl StorageResult {
+    pub fn key(&self) -> StorageKey {
+        StorageKey(self.digests.sha256.encode_hex())
```

Comment:
this is the convention I wanted to keep DRY, so no longer passing in a key, but only deriving it from the digests in a single place.

Reply:
Makes sense!